src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+		}
+		
+		if (eventDetail != null) {
+			updateEvent(cpDetail, eventDetail, event);
+		} else {
+			eventDetail = createEvent(cpDetail, event);
+		}
+		
+		eventsMap.put(event.getEventLabel(), eventDetail);
+	}
+
+	private CollectionProtocolEvent getEventFromDb(String cpShortTitle, String eventLabel) {
+		return daoFactory.getCollectionProtocolDao().getCpeByShortTitleAndEventLabel(cpShortTitle, eventLabel);
+	}
+	
+	private CollectionProtocolEventDetail createEvent(CollectionProtocolDetail cpDetail,
+			CollectionProtocolEventDetail event) throws Exception {
+		CollectionProtocolEventDetail eventDetail = new CollectionProtocolEventDetail();
+		eventDetail.setCpShortTitle(cpDetail.getShortTitle());
+		eventDetail.setEventLabel(event.getEventLabel());
+		
+		ResponseEvent<CollectionProtocolEventDetail> resp = cpSvc.addEvent(request(event));
+		resp.throwErrorIfUnsuccessful();
+		
+		return resp.getPayload();
+	}
+
+	private void updateEvent(CollectionProtocolDetail cpDetail, CollectionProtocolEventDetail eventDetail,
+			CollectionProtocolEventDetail event) {
+		
+	}
+	
+	//////////////////////////
+	//
+	// Collection Protocol
+	//
+	/////////////////////////
+	
+	private CollectionProtocolDetail checkCp(CollectionProtocolEventDetail event, CarsDetail object) throws Exception {
+		CollectionProtocolDetail cpDetail = new CollectionProtocolDetail();
+		cpDetail = cpMap.get(event.getCpShortTitle());
+		
+		if (cpDetail == null) {
+			cpDetail = getCpFromDB(event.getCpShortTitle());
+			if (cpDetail == null) {
+		        	cpDetail = createCp(object);
+			}
+			cpMap.put(event.getCpShortTitle(), cpDetail);
+		}
+		
+		return cpDetail;
+	}
+	
+	private CollectionProtocolDetail getCpFromDB(String cpShortTitle) {
+		CollectionProtocol cp = daoFactory.getCollectionProtocolDao().getCpByShortTitle(cpShortTitle);
+		return cp != null ? CollectionProtocolDetail.from(cp) : null;
+	}
+
+	private CollectionProtocolDetail createCp(CarsDetail input) throws Exception {
+		CollectionProtocolDetail cpDetail = new CollectionProtocolDetail();
+		cpDetail.setTitle(input.getIrbNumber());
+		cpDetail.setShortTitle(input.getIrbNumber());
+		setCpSites(input, cpDetail);
+		setCpPI(input, cpDetail);
+		
+		ResponseEvent<CollectionProtocolDetail> resp = cpSvc.createCollectionProtocol(request(cpDetail));
+		resp.throwErrorIfUnsuccessful();
+		
+		return resp.getPayload();
+	}
+
+	private void setCpPI(CarsDetail input, CollectionProtocolDetail cpDetail) {
+		UserSummary pi = new UserSummary();
+		pi.setEmailAddress(input.getPi());
+		cpDetail.setPrincipalInvestigator(pi);
+	}
+
+	private void setCpSites(CarsDetail input, CollectionProtocolDetail cpDetail) {
+		CollectionProtocolSiteDetail cpSite = new CollectionProtocolSiteDetail();
+		cpSite.setSiteName(input.getFacility());
+		
+		cpDetail.setCpSites(Arrays.asList(cpSite));
+	}
+	
+	private final static String SPECIMEN_STORAGE_TYPE_VIRTUAL = "Virtual";
+	
+	private final static String NOT_SPECIFIED = "Not Specified";
+}
```

Comment:
The order of methods in the file should be in the same order that they are first used and grouped together.

So this means, you will have 3 blocks

// Block 1: CP related private methods

// Block 2: CPE related private methods

// Block 3: SR related private methods.